src/dtls/mod.rs

```diff
@@ -55,15 +55,24 @@ impl ToString for Fingerprint {
         format!(
             "{} {}",
             self.hash_func,
-            self.bytes
-                .iter()
-                .map(|b| format!("{:02X}", b))
-                .collect::<Vec<_>>()
-                .join(":")
+            format_hex_colon_separated(&self.bytes),
         )
     }
 }
 
+pub(crate) fn format_hex_colon_separated(bytes: &[u8]) -> String {
+    // Each byte is 2 hexadecimal digits and colon - the last colon
+    let string_size = bytes.len() * 3 - 1;
+    let mut formatted = String::with_capacity(string_size);
+    for (i, byte) in bytes.iter().enumerate() {
+        if i > 0 {
+            formatted.push(':');
+        }
+        formatted.push_str(&format!("{:02X}", byte));
+    }
+    formatted
+}
+
```

Comment:
I suspect the fmt API the Display trait is even more optimized, not using an intermediate String. But I don't really care since the serialization of this happens so rarely.

Reply:
switched back to Display trait